registrations/tasks.py

```diff
@@ -43,3 +44,41 @@ def send_registration_to_openhim(
     )
     response.raise_for_status()
     return (response.status_code, response.headers, response.content)
+
+
+@app.task(
+    autoretry_for=(RequestException, SoftTimeLimitExceeded),
+    retry_backoff=True,
+    max_retries=15,
+    acks_late=True,
+    soft_time_limit=10,
+    time_limit=15,
+)
+def send_registration_to_rapidpro(
+    contact, msisdn, referral_msisdn, channel, clinic_code, timestamp
+):
+    # Create/Update contact
+    group_name = "nurseconnect-%s" % channel.lower()
+    group = tembaclient.get_groups(name=group_name).first()
+    contact_data = {
+        "preferred_channel": channel.lower(),
+        "registered_by": referral_msisdn,
+        "facility_code": clinic_code,
+        "registration_date": datetime.fromtimestamp(timestamp).isoformat(),
+    }
+    if contact:
+        uuid = contact.get("uuid")
```

Comment:
I think we should probably always get the contact fresh from RapidPro in this task.

That way, if we create the contact, and one of the consecutive requests fail, we don't try to create a duplicate contact (which will fail), because we still think that the contact doesn't exist.